libtransmission/transmission.h

```diff
@@ -1397,6 +1405,22 @@ void tr_torrentAmountFinished(tr_torrent const* torrent, float* tab, int n_tabs)
  */
 void tr_torrentVerify(tr_torrent* torrent, bool force = false);
 
+#ifdef __cplusplus
+
+struct tr_verify_file_result
+{
+    std::string_view filename;
+    uint64_t size;
+    std::vector<std::string> errors;
+    bool fail;
+};
```

Comment:
:thinking: 

Instead of having this in the public API, what would you think about `tr_verify_worker::verify_torrent()` taking a third argument that is an error callback, e.g.

```c++
using tr_verify_error_cb = void(*)(
    std::string_view filename,
    uint64_t size,
    std::string_view error);

...

[[nodiscard]] bool tr_torrentImmediateVerify(tr_torrent* torrent, tr_verify_error_cb cb = nullptr);
```

IMO this would be more flexible and have a smaller public API footprint

Reply:
I was on the fence with using a callback vs passing a vector when I wrote the code. Using a vector seemed to be easier. But I'm happy to change it to a callback. An advantage of using the callback is if `--unsorted` is used, we can output the status immediately instead of accumulating it and then outputting it at the end. This is useful with huge torrents where you could be waiting a while before you see the files listed.